Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -284,6 +284,44 @@ extension NIOSSLCertificate: Equatable {
     }
 }
 
+
+extension NIOSSLCertificate: Hashable {
+    /// This is a silly thunk type that exists just to provide stack-allocation for a SHA256 digest.
+    private struct SHA256Digest {
+        // A SHA256 digest is 256 bits long, or 32 bytes, or 4 UInt64s. We store this
+        // as UInt64s here, but they are always accessed as UInt8s.
+        private var storage: (UInt64, UInt64, UInt64, UInt64)
+
+        init() {
+            self.storage = (0, 0, 0, 0)
+        }
+
+        mutating func withUnsafeMutablePointer<T>(_ body: (UnsafeMutableBufferPointer<UInt8>) throws -> T) rethrows -> T {
+            return try Swift.withUnsafeMutablePointer(to: &self.storage) { (storagePtr: UnsafeMutablePointer<(UInt64, UInt64, UInt64, UInt64)>) -> T in
+                return try storagePtr.withMemoryRebound(to: UInt8.self, capacity: 1) { bytesPtr -> T in
```

Comment:
`capacity: 1` isn't right here, should be `<(UInt64, ...)>.size`, right?

Reply:
I honestly don't know. The [docs](https://developer.apple.com/documentation/swift/unsafepointer/2430863-withmemoryrebound) say:

> **`count`**
>   The number of instances of `Pointee` to bind to `type`.

Here, `type` is the name of the argument with the label `to`, which implies that I don't want to say how many of `to` I want, I want to say how many of the original elements I am binding, but I don't think it's very clear. @atrick, can you clarify?

Reply:
The docs have this note:
```
  /// - Note: Only use this method to rebind the pointer's memory to a type
  ///   with the same size and stride as the currently bound `Pointee` type.
  ///   To bind a region of memory to a type that is a different size, convert
  ///   the pointer to a raw pointer and use the `bindMemory(to:capacity:)`
  ///   method.
```
See [SR-11082: Support withMemoryRebound(to:) with mismatched sizes](https://bugs.swift.org/browse/SR-11082).

Until then, the most "proper" thing to do would be call bind memory yourself, or create your own withMemoryRebound wrapper for it:

```
let toPtr = UnsafeMutableRawPointer(fromPtr).bindMemory(to: ToType.self, capacity: numElementsOfToType)

// do some stuff

UnsafeMutableRawPointer(toPtr).bindMemory(to: FromType.self, capacity: numElementsOfFromType)
```

Reply:
Huh, weird. Thanks for weighing in, I’ll update the patch accordingly!